core/services/webhook/external_initiator_manager.go

```diff
@@ -86,44 +87,50 @@ func (m externalInitiatorManager) Notify(js models.JobSpec) error {
 
 // NotifyV2 sends a POST notification to the External Initiator
 // responsible for initiating the Job Spec.
-func (m externalInitiatorManager) NotifyV2(
-	jobID uuid.UUID,
-	initrName string,
-	initrSpec *models.JSON,
-) error {
-	ei, err := m.FindExternalInitiatorByName(initrName)
+func (m externalInitiatorManager) NotifyV2(webhookSpecID int32) error {
+	eiWebhookSpecs, jobID, err := m.Load(webhookSpecID)
 	if err != nil {
-		return errors.Wrap(err, "external initiator")
-	} else if ei.URL == nil {
-		return nil
-	} else if initrSpec == nil {
-		return errors.New("body must be defined")
-	}
-
-	notice := JobSpecNoticeV2{
-		JobID:  jobID,
-		Type:   initrName,
-		Params: *initrSpec,
-	}
-	buf, err := json.Marshal(notice)
-	if err != nil {
-		return errors.Wrap(err, "new Job Spec notification")
-	}
-	req, err := newNotifyHTTPRequest(buf, ei)
-	if err != nil {
-		return errors.Wrap(err, "creating notify HTTP request")
+		return err
 	}
-	resp, err := http.DefaultClient.Do(req)
-	if err != nil {
-		return errors.Wrap(err, "could not notify '%s' (%s)")
-	}
-	defer logger.ErrorIfCalling(resp.Body.Close)
-	if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
-		return fmt.Errorf(" notify '%s' (%s) received bad response '%s'", ei.Name, ei.URL, resp.Status)
+	for _, eiWebhookSpec := range eiWebhookSpecs {
+		ei := eiWebhookSpec.ExternalInitiator
+		if ei.URL == nil {
+			continue
+		}
+		notice := JobSpecNoticeV2{
+			JobID:  jobID,
+			Type:   ei.Name,
+			Params: eiWebhookSpec.Spec,
+		}
+		buf, err := json.Marshal(notice)
+		if err != nil {
+			return errors.Wrap(err, "new Job Spec notification")
+		}
+		req, err := newNotifyHTTPRequest(buf, ei)
+		if err != nil {
+			return errors.Wrap(err, "creating notify HTTP request")
+		}
+		resp, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return errors.Wrap(err, "could not notify '%s' (%s)")
+		}
+		defer logger.ErrorIfCalling(resp.Body.Close)
+		if !(resp.StatusCode >= 200 && resp.StatusCode < 300) {
+			return fmt.Errorf(" notify '%s' (%s) received bad response '%s'", ei.Name, ei.URL, resp.Status)
+		}
 	}
 	return nil
 }
 
+func (m externalInitiatorManager) Load(webhookSpecID int32) (eiWebhookSpecs []job.ExternalInitiatorWebhookSpec, jobID uuid.UUID, err error) {
+	row := m.db.Raw("SELECT external_job_id FROM jobs WHERE webhook_spec_id = ?", webhookSpecID).Row()
```

Comment:
hm worth creating a webhook/orm.go file for mockability?

Reply:
Since ExternalInitiatorManager is already mockable, I think that would be overkill.